fpdf/html.py

```diff
@@ -334,8 +338,11 @@ def __init__(
         self.align = ""
         self.style_stack = []  # list of FontFace
         self.indent = 0
+        self.line_height_stack = []
         self.ol_type = []  # when inside a <ol> tag, can be "a", "A", "i", "I" or "1"
         self.bullet = []
+        if list_vertical_margin is None:
+            self.list_vertical_margin = 0.3 / self.pdf.k
```

Comment:
This converts 0.3 points (0.12 mm) to document units.

You've may have seen `self.font_size` being converted like this a few times. The pitfall being that `self.font_size` is derived from `pdf.font_size_pt`, which actually is in points. We should probably rename `self.font_size` in this module also to `self.font_size_pt` to make this more obvious.

Besides the "magical number" problem in this specific instance, there are more cases where you do a similarly incorrect conversion.

Reply:
How is this conversion incorrect, and in what sense is it a 'magical number' if it's simply the default value for the vertical margin?

Reply:
Ah, so the conversion is intentional, but then the amount doesn't look like something you'd want.

The default here used to be 2 mm (implemented incorrectly without taking document units into account).

If you wanted to specify the same amount in points and convert it to document units, then `2 * util.get_scale_factor("mm")` would result in ~5.67 points, not 0.3.
Your current value amounts to a tenth of a mm, so you could just as well leave the margin away.

Reply:
> And again, a constant like that should come with a comment specifying its unit and how much that is in mm (most people are not aware of how big a typographical point is)

Given that there were no such comments prior to my intrusion, I assumed that I shouldn't be changing that. Where else should I introduce such comments, if you don't mind me asking?

> And no, I don't see this depending on any other size. What did you have in mind?

I am/was confused about why we should make `get_scale_factor()` calls instead of just using `self.pdf.k`. Do I understand it correctly now that you want me to return the relevant default values to how they were prior to my changes, and to convert them with `* get_scale_factor('mm') / self.pdf.k` into the appropriate units?

Reply:
> Given that there were no such comments prior to my intrusion, I assumed that I shouldn't be changing that. Where else should I introduce such comments, if you don't mind me asking?

This module still contains a lot of legacy code of rather low quality. Any improvement to that is welcome. So far we have just fixed the parts we were touching anyway when adding features. I recommend you do the same, in order to keep the focus on your actual goal. At some later point someone will have to go through every line with a fine toothed comb, but that is a separate and rather tedious task, requiring a good grasp of the whole code base.

I've commented on recommended unit use further above.